src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -3058,15 +3063,17 @@ func LaunchCustomQemu(ctx context.Context, path string, params []string, fds []*
 
 	cmd.SysProcAttr = attr
 
-	var stderr bytes.Buffer
-	cmd.Stderr = &stderr
+	reader, err := cmd.StderrPipe()
+	if err != nil {
+		logger.Errorf("Unable to connect stderr to a pipe")
+		return nil, nil, err
+	}
 	logger.Infof("launching %s with: %v", path, params)
 
-	err := cmd.Run()
+	err = cmd.Start()
 	if err != nil {
 		logger.Errorf("Unable to launch %s: %v", path, err)
-		errStr = stderr.String()
-		logger.Errorf("%s", errStr)
+		return nil, nil, err
```

Comment:
Why don't you return the reader in that case? Do we assume that if we failed to start, there is nothing on `stderr`?

Reply:
Yes. AFAICT from the source, the go library doesn't write anything on stderr. Moreover it closes all the pipes on errors. I can't think of anything else writing to file descriptor 2.